self-core-impl/src/main/java/com/selfxdsd/core/GitlabRepo.java

```diff
@@ -54,6 +54,24 @@
         super(resources, uri, owner, storage);
     }
 
+    /**
+     * Creates a Gitlab Repo from name.
+     * @param resources Gitlab's JSON Resources.
+     * @param repoName Repository name.
+     * @param owner Owner of this repo.
+     * @param storage Storage used to save the Project when
+     *  this repo is activated.
+     * @return GitlabRepo.
+     */
+    static GitlabRepo createFromName(final String repoName,
+                                     final JsonResources resources,
+                                     final User owner,
+                                     final Storage storage) {
+        final URI repo = URI.create("https://gitlab.com/api/v4/projects/"
+            + repoName);
+        return new GitlabRepo(resources, repo, owner, storage);
+    }
```

Comment:
@criske Why is this method needed?

Reply:
Same logic for uri creation is used in different places (`Gitlab` and `GitlabOrganizationRepos`). But the main inconvenient is that Gitlab API doesn't give full api-url for repos in their json responses as Github does. So each time we need to extract the repo name from json's `path`-key  and from there build the Uri used by `GitlabRepo`.